### ISSUE TITLE
Massive lag in network + AI game (~5FPS)

### ISSUE THREAD

**Issue by DeeeeLAN** [MustaphaTR/Romanovs-Vengeance#88]
Hi, I am not sure how to go about helping debug this issue. There were three of us playing against 4 AI over the network. All of us using the most recent release (7/7/22). For a significant amount of the game, it was crawling at a super low FPS. My assumption was there were too many units and structures and it was beyond what the engine could handle. 

One possibility I was noticing that was kind of weird was I had built 10-15 dreadnoughts who were bombarding one of the enemy's bases, and the mini-map was constantly spamming red triangles in the area, I think every time one of the dreadnought missiles was shot down it was showing as a character lost on the mini-map.

Near the end after we had taken out 2-3 of the enemy bases the FPS picked back up a bit. I was also using the dreadnoughts a lot less at that point. 

Are there any logs or anything I can share that could help diagnose?

**Comment by MustaphaTR** [comment]
Which map was that? There's a known lag issue that happens when units try to path to an area accessable by low bridges that are currently destroyed. Heidelberg is a map that this can happen, but that's a 2p map so can't be the map you had the issue with, but can still be the same thing.

**Comment by DeeeeLAN** [comment]
Ah, that must have been it then, we were playing Arctic Circle and definitely had destroyed bridges all over the place.

**Comment by MustaphaTR** [comment]
Yeah, Arctic Circle is a map known for this issue. That was most likely it.

**Comment by DeeeeLAN** [comment]
We will try again with a non-bridge map and hopefully things work better. For reference, where are the logs and replay files saved? I am on Linux. I will grab them next time we have an issue so I can submit them if needed.

**Comment by MustaphaTR** [comment]
Log files should be in `~/.config/openra/logs` on Linux.

Also note that this is only an issue with low bridges, high bridges aren't even destroyable at the first place currently.

**Comment by DeeeeLAN** [comment]
[replay.tar.gz](https://github.com/MustaphaTR/Romanovs-Vengeance/files/9223034/replay.tar.gz)

Here is the replay, hope it helps.

**Comment by bradleysimard** [comment]
Can we get another compiled release soon?

The AI lag makes matches unplayable.

**Comment by dnqbob** [comment]
> Can we get another compiled release soon?
> 
> The AI lag makes matches unplayable.
> 
> Also I can tell you it's not specifically related to bridges that are destroyed. I think it has more to do with water units.

You are correct, units being stuck by enclosed immovable actors can cause greater lags. see https://github.com/OpenRA/OpenRA/issues/20948. We are trying to fix that.

**Comment by MustaphaTR** [comment]
I have obeserved an AI game with all bridges on Arctic Circle with all bridges destroyed. The FPS did fall later on but not to the unplayably slow ~5 FPS level that it used to get but to around ~20 FPS at worst. I assume this was fixed in the pathfinder reworks that has been done since then. So closing the issue.
